californium-core/src/test/java/org/eclipse/californium/core/coap/RequestTest.java

```diff
@@ -67,8 +68,21 @@ public void testSetURISetsDestination() throws UnknownHostException {
 
 	@Test
 	public void testSetURISetsUriHostOption() throws UnknownHostException {
-		Request req = Request.newGet().setURI("coap://iot.eclipse.org");
-		assertThat(req.getOptions().getUriHost(), is("iot.eclipse.org"));
```

Comment:
maybe "localhost" and "127.0.0.1" are mapped trivial, "iot.eclipse.org" to its IP address not. 
So, with "127.0.0.1" as address, the "URI HOST" doesn't provide more information.
With "???.???.???.???" as address, the "URI HOST" provides more information.